features/storage/kvstore/global_api/kvstore_global_api.cpp

```diff
@@ -0,0 +1,215 @@
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    return kv_instance->get(key, buffer, buffer_size, actual_size);
+}
+
+int kv_get_info(const char *full_name_key, kv_info_t *info)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    KVStore::info_t inner_info;
+    ret =  kv_instance->get_info(key, &inner_info);
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+    info->flags = inner_info.flags;
+    info->size =  inner_info.size;
+    return ret;
+}
+
+int kv_remove(const char *full_name_key)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_name_key, &kv_instance, key);
+    return kv_instance->remove(key);
+}
+
+int kv_set_start(kv_set_handle_t *handle, const char *key, size_t final_data_size, uint32_t create_flags)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    (*handle) = new _opaque_kv_set_handle;
+    if (*handle == NULL) {
+        return MBED_ERROR_FAILED_OPERATION;
+    }
+
+    KVStore *kv_instance = NULL;
+    KVStore::set_handle_t *inner_handle = new KVStore::set_handle_t;
+    char out_key[KV_MAX_KEY_LENGTH];
+    kv_lookup(key, &kv_instance, out_key);
+    ret = kv_instance->set_start(inner_handle, out_key, final_data_size, create_flags);
+    if (MBED_SUCCESS != ret) {
+        delete inner_handle;
+        return ret;
+    }
+
+    (*handle)->set_handle = inner_handle;
+    (*handle)->handle_is_open = true;
+
+    return ret;
+}
+
+int kv_set_add_data(kv_set_handle_t handle, const void *value_data, size_t data_size)
+{
+    if (!handle->handle_is_open) {
+       return MBED_ERROR_INVALID_ARGUMENT;
+    }
+
+    return handle->kvstore_intance->set_add_data(*handle->set_handle, value_data, data_size);
+}
+
+int kv_set_finalize(kv_set_handle_t handle)
+{
+    if (!handle->handle_is_open) {
+        return MBED_ERROR_INVALID_ARGUMENT;
+    }
+
+    int ret = handle->kvstore_intance->set_finalize(*handle->set_handle);
+
+    delete handle->set_handle;
+    delete handle;
+
+    return ret;
+}
+
+int kv_iterator_open(kv_iterator_t *it, const char *full_prefix)
+{
+    int ret = storage_configuration();
+    if (MBED_SUCCESS != ret) {
+        return ret;
+    }
+
+    (*it) = new _opaque_kv_key_iterator;
+    if (*it == NULL) {
+        return MBED_ERROR_FAILED_OPERATION;
+    }
+
+    KVStore *kv_instance = NULL;
+    char key[KV_MAX_KEY_LENGTH];
+    kv_lookup(full_prefix, &kv_instance, key);
+    (*it)->kvstore_intance = kv_instance;
+
+    KVStore::iterator_t *inner_it = new KVStore::iterator_t;
```

Comment:
Not very important, but for good practice - better set `iterator_is_open` to false after allocation, in case of failures.